README.md

```diff
@@ -1,105 +1,366 @@
+older of Font Awesome. You might head over there to learn about how it works.
+
+> https://fontawesome.com/how-to-use/svg-with-js
+
+#### Going from an older pre-release version?
+
+See [UPGRADING.md](./UPGRADING.md).
+
+You might also be interested in the larger umbrella project [UPGRADING.md](https://github.com/FortAwesome/Font-Awesome/blob/master/UPGRADING.md)
+
+## Installation
+
+```
+$ yarn add @fortawesome/fontawesome-svg-core
+$ yarn add @fortawesome/free-solid-svg-icons
+$ yarn add @fortawesome/angular-fontawesome
+```
+
+## Add more styles or Pro icons
+
+Brands are separated into their own style and for customers upgrading from
+version 4 to 5 we have a limited number of Regular icons available.
+
+**Visit [fontawesome.com/icons](https://fontawesome.com/icons) to search for free and Pro icons**
+
+```
+$ yarn add @fortawesome/free-brands-svg-icons
+$ yarn add @fortawesome/free-regular-svg-icons
+```
+
+If you are a [Font Awesome Pro](https://fontawesome.com/pro) subscriber you can install Pro packages.
+
+```
+$ yarn add @fortawesome/pro-solid-svg-icons
+$ yarn add @fortawesome/pro-regular-svg-icons
+$ yarn add @fortawesome/pro-light-svg-icons
+```
+
+Using the Pro packages requires [additional configuration](https://fontawesome.com/how-to-use/js-component-packages).
+
+## Usage
+
+These examples are based on a freshly created project with [Angular CLI].
+
+[Angular CLI]: https://cli.angular.io
+
+### Explicit reference
+
+Not as convenient as using the library but if you believe "explicit is better than
+implicit" then this method if for you.
+
+`src/app/app.component.html`
+
+```html
+<div style="text-align:center">
+  <fa-icon [icon]="faCoffee"></fa-icon>
+</div>
 ```
-<fa-icon [icon]="faUser"></fa-icon>
+
+`src/app/app.component.ts`
+
+```typescript
+import { Component } from '@angular/core';
+import { faCoffee } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  title = 'app';
+  faCoffee = faCoffee;
+}
 ```
 
-Run `yarn start` or `npm run start` to see more example uses.
+`src/app/app.module.ts`
 
-## Advanced Usage
+1. Import `{ FontAwesomeModule } from '@fortawesome/angular-fontawesome'`
+1. Add `FontAwesomeModule` to `imports`
 
-*With Mask and Transform*
+```typescript
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+
+import { AppComponent } from './app.component';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    FontAwesomeModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
 ```
-<fa-icon [icon]="faCircle" transform="shrink-9 right-4" [mask]="faSquare"></fa-icon>
+
+### Using the Font Awesome Library
```

Comment:
I was confused when I started reading this section. I initially thought that by "Font Awesome Library" you meant the API library, as in `fontawesome-svg-core`. I re-read the first part about templates 3 times before I realized you were talking about the `library.add()` library. I wonder if that's going to trip up anyone else? I guess, without thinking really thinking about it, I've started calling `fontawesome-svg-core` the "base" or "API" library. And the `library.add()` thing the "icon library". And I suppose a newbie might also wonder if `free-solid-svg-icons` isn't a library, but I normally call that an "icon package" or "icon pack".

So I'm calling this out as something I've been taking for granted now made strange, forcing me to maybe see it through a newcomer's eyes.

For the moment, I'm going to push a commit that changes this to "Icon Library". But I'll grant that this its debatable what stuff should be named. So, pushback or whatever.